### ISSUE TITLE
Root Result with BBox Issue.

### ISSUE THREAD

**Issue by YangJae96** [mks0601/3DMPPE_ROOTNET_RELEASE#23]
Hello.

Thank you for your great work.

I tried the demo.py in the RootNet with your "input.jpg".
You gave us the bbox values like this
bbox_list = [
 [139.41, 102.25, 222.39, 241.57],\
 [287.17, 61.52, 74.88, 165.61],\
 [540.04, 48.81, 99.96, 223.36],\
 [372.58, 170.84, 266.63, 217.19],\
[0.5, 43.74, 90.1, 220.09]
]  and this seems to get good Root results.

But there is a problem when I use my bbox results that I obtained from Detectron2.
bbox_list = [[367.4241, 177.2487, 636.8704, 389.0179],
        [169.7263, 103.9277, 365.6891, 341.9860],
        [  1.0619,  43.5408,  88.0144, 266.1570],
        [537.6930,  51.6290, 639.7495, 265.5615],
        [292.3224,  59.8680, 364.7124, 224.1436]]
![output](https://user-images.githubusercontent.com/57695621/90800128-2d4a7f80-e34f-11ea-916b-ce21332f1336.jpg)

![output_root_2d_4](https://user-images.githubusercontent.com/57695621/90800670-ec9f3600-e34f-11ea-8beb-a51cb6538d31.jpg)
![output_root_2d_1](https://user-images.githubusercontent.com/57695621/90800702-f6289e00-e34f-11ea-83c1-f247c3b7d2b8.jpg)

The results of the RootNet is bad. I have also tested with other images like 2 people in image
but I keep getting bad results although I have detected the humans.
I have used all the pretrained Detectron Mask RCNN and Faster RCNN Models.
and  I also used  the pretrained model of the RootNet   "snapshot18.pth" that you provided. 

Is The bbox you provided a Ground truth?? I can't figure out what the issue is.
Can I ask you some advice please?

Thank you.

**Comment by mks0601** [comment]
Hi,

I checked the bbox you gave, however, it seems they are wrong. Below images are just cropped images from `input.jpg` with your bbox. As you can see some images are not properly cropped.
![0](https://user-images.githubusercontent.com/13601723/90833221-8a135d80-e382-11ea-95d1-f99e805d61e0.jpg)
![1](https://user-images.githubusercontent.com/13601723/90833223-8b448a80-e382-11ea-8648-5cb05d721da8.jpg)
![2](https://user-images.githubusercontent.com/13601723/90833224-8bdd2100-e382-11ea-990f-6771d3cd43f8.jpg)
![3](https://user-images.githubusercontent.com/13601723/90833225-8c75b780-e382-11ea-99f5-8931198d5c09.jpg)
![4](https://user-images.githubusercontent.com/13601723/90833227-8c75b780-e382-11ea-9f7e-dbf119816740.jpg)

And below images are RootNet results. 
![output_root_2d_0](https://user-images.githubusercontent.com/13601723/90833301-b3cc8480-e382-11ea-8380-d3e13d60da0d.jpg)
![output_root_2d_1](https://user-images.githubusercontent.com/13601723/90833307-b4fdb180-e382-11ea-954d-5952e83aa988.jpg)
![output_root_2d_2](https://user-images.githubusercontent.com/13601723/90833309-b62ede80-e382-11ea-9025-e9d06262bf5f.jpg)
![output_root_2d_3](https://user-images.githubusercontent.com/13601723/90833312-b6c77500-e382-11ea-8ad6-e2b20488a1bf.jpg)
![output_root_2d_4](https://user-images.githubusercontent.com/13601723/90833314-b7600b80-e382-11ea-857c-509873faa6ee.jpg)
As you can see, if the bbox is correct, it produces good results. 

Similar experimental results in my paper (Table 2) show that human detection performance affect marginally on human root joint localization and 3D root-relative pose estimation.

**Comment by YangJae96** [comment]
Thank you for your quick reply!!
The problem was the bbox coordinates I used from detectron2 mask-rcnn was
(xmin,ymin, xmax,ymax)...  lefttop coord and right bottom coord.
which didn't match with your bbox coord (xmin,ymin,width,height)

Thank you again for great work!

**Comment by vigneshrk29** [comment]
Hi,

Sorry to reopen this issue but would width and height be equal to xmax-xmin and ymax-ymin?

Thanks

**Comment by mks0601** [comment]
which width and height do you mean?

**Comment by vigneshrk29** [comment]
Thanks for the quick reply. In the bbox values, they mentioned that instead of [xmin, ymin, xmax, ymax] you are meant to put [xmin, ymin, width, height]. I wanted to know if width and height was the difference between max and min values?

**Comment by vigneshrk29** [comment]
I ran the pose net and rootnet using a image from neither of the datatsets. I got bbox values from yolov5. This is the ouput I get:
![root](https://user-images.githubusercontent.com/84029062/236648486-f8f3e86f-26f1-4737-b26b-b9f17660c461.JPG)
![3dPose](https://user-images.githubusercontent.com/84029062/236648494-73f503e6-bff8-4ea8-bb21-46a9856c850d.JPG)

The root looks good but the 3dpose isnt great. So not sure where I am going wrong

**Comment by mks0601** [comment]
You're correct. The height and width represent xmax-xmin and ymax-ymin, respectively. Although that pose seems easy, contact between different body part like contact between hand and body of that image is one of the biggest challenge in 3D human pose community because of the depth ambiguity. As you might note, 2D-projected 3D pose seems reasonable.

**Comment by vigneshrk29** [comment]
Yes, the 2d-projected pose is reasonble. Thank you. 
Brilliant work btw
